tools/project_manager/editproject.php

```diff
@@ -230,10 +232,9 @@ public function set_from_post()
                 return _("You are not authorized to manage this project.").": '$this->projectid'";
             }
         } elseif (isset($_POST['clone_projectid'])) {
-            // we're creating a clone
+            // we're creating a clone; initialize our object with the original
+            // project, then override with edits from $_POST (below)
             $clone_projectid = get_projectID_param($_POST, 'clone_projectid');
-            $this->clone_projectid = $clone_projectid;
-
             $this->set_from_clone($clone_projectid);
         } else {
             $this->project = new Project();
```

Comment:
Just curious about this -- if the purpose is to clone a project, do we want this here? Or should there be some kind of exception thrown because the project that whoever was trying to clone is a nonexistent project? (The way it's done obviously deals with that, so it's not like it's necessary to do anything -- I'm just curious.)

Reply:
This else block is the case where a user is creating a new project (not editing an existing one and not cloning one). Only the elseif above it deals with cloning in this section of `set_from_post()`. Does that make sense?

Reply:
Absolutely. I just didn't understand that the function was multi-use.